Sources/SwiftSyntax/Raw/RawSyntaxTokenView.swift

```diff
@@ -204,8 +202,21 @@ public struct RawSyntaxTokenView {
     arena.addChild(self.raw.arenaReference)
     switch raw.rawData.payload {
     case .parsedToken(var payload):
-      payload.presence = newValue
-      return RawSyntax(arena: arena, payload: .parsedToken(payload))
+      if arena == self.raw.arenaReference {
+        payload.presence = newValue
+        return RawSyntax(arena: arena, payload: .parsedToken(payload))
+      }
+      // If the modified token is allocated in a different arena, it might have
+      // a different or no `parseTrivia` function. We thus cannot use a
+      // `parsedToken` anymore.
+      return .makeMaterializedToken(
+        kind: formKind(),
+        leadingTrivia: formLeadingTrivia(),
+        trailingTrivia: formTrailingTrivia(),
+        presence: newValue,
+        tokenDiagnostic: tokenDiagnostic,
+        arena: arena
+      )
```

Comment:
Just to make sure I understand correctly, this should be reverted when `SwiftParser` is merged, right?

Reply:
I’d prefer to keep it even if/when we merge SwiftSyntax + SwiftParser. Technically, the old implementation is not correct because the new arena could have a different `parseTriviaFunction` than the arena the node is allocated in and thus the trivia pieces could change here if we don’t materialize them.